Competition2017/src/competition/subsystems/drive/commands/DriveForDistanceCommand.java

```diff
@@ -26,14 +27,20 @@
 
     private double previousPositionInches;
     
+    private final PIDManager headingDrivePid;
+    private ContiguousHeading targetHeading;
+    public final double defaultPValue = 1/80d;
+
     private int onTargetCount = 0;
     
     @Inject
     public DriveForDistanceCommand(DriveSubsystem driveSubsystem, XPropertyManager propManager) {
         this.driveSubsystem = driveSubsystem;
         this.requires(driveSubsystem);
         this.travelManager = new PIDManager("Drive to position", propManager, 0.1, 0, 0, 0.5, -0.5);
-        
+
+        headingDrivePid = new PIDManager("HeadingModule", propManager, defaultPValue, 0, 0);
+        targetHeading = new ContiguousHeading();
         onTargetCountThresholdProp = propManager.createPersistentProperty("DrvToPos min stabilization time", 3);
```

Comment:
In general, the contract when defining an `execute` method is that it will be called consistently, once per loop; that makes the current PID implementation work, although it would preferably be improved. On the other hand, there's no such expectation with `isFinished`: it's a getter, and getters should never have side-effects. If it does, it's no longer a getter. This is most impactful in tests, where we are manually calling these methods. In a test case, it makes a lot more sense to a reader to call it once and assert, change the elapsed time, and assert again than to repeatedly call the `isFinished` method until it changes its mind.

If this were inside the `PIDManager` or the `PID` class, I'd stick with number of calls to `execute` as that is what the rest of the PID logic uses. However, here, this has no attachment to the mechanisms of the PID logic: it should use time if that's what makes the most sense in the context of its task.